include/mxx/distribution.hpp

```diff
@@ -111,15 +111,18 @@ void stable_distribute(_InIterator begin, _InIterator end, _OutIterator out, con
         size_t prefix = mxx::exscan(local_size, comm);
 
         // calculate where to send elements
+        // if there are any elements to send
         std::vector<size_t> send_counts(comm.size(), 0);
-        blk_dist part(total_size, comm.size(), comm.rank());
-        int first_p = part.rank_of(prefix);
```

Comment:
This is the line that leads to an error if `prefix` is the same as the `total_size`.